src/main/kotlin/no/nav/familie/ef/sak/selvstendig/NæringsinntektKontrollService.kt

```diff
@@ -31,44 +33,46 @@ class NæringsinntektKontrollService(
     val vedtakService: VedtakService,
     val sigrunService: SigrunService,
     val inntektService: InntektService,
+    val andelsHistorikkService: AndelsHistorikkService,
+    val tilkjentYtelseService: TilkjentYtelseService,
 ) {
     private val secureLogger = LoggerFactory.getLogger("secureLogger")
     private val årstallIFjor = YearMonth.now().minusYears(1).year
     private val månedsperiodeIFjor = Månedsperiode(YearMonth.of(årstallIFjor, 1), YearMonth.of(årstallIFjor, 12))
 
     fun sjekkNæringsinntektMotForventetInntekt(): List<UUID> {
-        val behandlingIds = mutableListOf<UUID>()
+        val fagsakIds = mutableListOf<UUID>()
         if (LeaderClient.isLeader() == true) {
             val oppgaver = hentOppgaverForSelvstendigeTilInntektskontroll()
 
             oppgaver.forEach { oppgave ->
                 val personIdent =
                     oppgave.identer?.firstOrNull { it.gruppe == IdentGruppe.FOLKEREGISTERIDENT }?.ident
                         ?: oppgave.personident ?: throw Exception("Fant ikke registrert ident på oppgave ${oppgave.id}")
-
                 secureLogger.info("Kontrollerer person med ident: $personIdent")
 
-                val fagsakOS = fagsakService.finnFagsaker(setOf(personIdent)).filter { it.stønadstype == StønadType.OVERGANGSSTØNAD }.firstOrNull() ?: throw RuntimeException("Fant ikke fagsak for overgangsstønad for person: $personIdent")
-                val behandling = behandlingService.finnSisteIverksatteBehandling(fagsakOS.id) ?: throw RuntimeException("Fant ikke behandling for fagsakId: ${fagsakOS.id}")
-                val antallMånederMedVedtakForFjoråret = antallMånederMedVedtakForFjoråret(behandling)
+                val fagsakOS = fagsakService.finnFagsaker(setOf(personIdent)).firstOrNull { it.stønadstype == StønadType.OVERGANGSSTØNAD } ?: throw RuntimeException("Fant ikke fagsak for overgangsstønad for person: $personIdent")
+                val behandlingId = behandlingService.finnSisteIverksatteBehandling(fagsakOS.id)?.id ?: throw RuntimeException("Fant ingen gjeldende behandling for fagsakId: ${fagsakOS.id}")
+                val tilkjentYtelse = tilkjentYtelseService.hentForBehandling(behandlingId)
 
+                val antallMåneder = antallMånederMedVedtakForFjoråret(tilkjentYtelse)
                 val næringsinntekt = hentFjoråretsNæringsinntekt(fagsakOS.fagsakPersonId)
 
-                if (antallMånederMedVedtakForFjoråret > 3 && næringsinntekt > INNTEKTSGRENSE_FOR_KONTROLL_AV_AKTIVITET) {
+                if (antallMåneder > 3 && næringsinntekt > INNTEKTSGRENSE_FOR_KONTROLL_AV_AKTIVITET) {
                     val fjoråretsPersoninntekt = inntektService.hentÅrsinntekt(personIdent, årstallIFjor)
                     secureLogger.info("Forrige års inntekt for person uten ytelse fra offentlig: $fjoråretsPersoninntekt")
                     if (fjoråretsPersoninntekt == 0) {
```

Comment:
Skjønner at vi her kun sjekker på de som har 0 i personinntekt, men hadde det vært nyttig å logge en annen info om de som har `personInntekt+næringsinntekt > fi * 1.1` også? 

Eller er det andre mekanismer for å fange opp disse?

Reply:
Høres ut som en fin avklaring jeg ta med funksjonelle:)